### PR TITLE
Aws secret manager

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Add AWS Secret Manager as a supported secret injector in flyte's pod webhook.

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [X] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1104

### CODE REVIEW

**Review by wild-endeavor — DISMISSED** [review]

File: pkg/webhook/utils.go [hunk 1]
```diff
@@ -60,17 +60,17 @@ func CreateVolumeMountForSecret(volumeName string, secret *core.Secret) corev1.V
...
-		c.VolumeMounts = append(c.VolumeMounts, mount)
+		c.VolumeMounts = appendVolumeMountIfNotExists(c.VolumeMounts, mount)
 		res = append(res, c)
 	}
 
 	return res
 }
 
-func UpdateEnvVars(containers []corev1.Container, envVar corev1.EnvVar) []corev1.Container {
+func AppendEnvVars(containers []corev1.Container, envVar corev1.EnvVar) []corev1.Container {
```
Comment (kumare3) [inline pkg/webhook/utils.go:73]: we have this method in pluginmachinery already right?

File: pkg/webhook/aws_secret_manager.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	default:
+		err := fmt.Errorf("unrecognized mount requirement [%v] for secret [%v]", secret.MountRequirement.String(), secret.Key)
+		logger.Error(ctx, err)
+		return p, false, err
+	}
+
+	return p, true, nil
+}
+
+func createAWSSidecarContainer(cfg config.AWSSecretManagerConfig, p *corev1.Pod, secret *core.Secret) corev1.Container {
```
Comment (kumare3) [inline pkg/webhook/aws_secret_manager.go:122]: is this a sidecar or an init container. If sidecar, isnt there a race condition, between the mounting and the user container trying to access the secret?

**Comment by kumare3** [comment]
we should have probably created webhook as a top level folder instead of under pkg? couple questions and then +1

File: pkg/webhook/utils.go (hunk 1 above)
Comment (EngHabu) [reply pkg/webhook/utils.go:73]: Couldn't find that... can you post a link?

Comment (kumare3) [reply pkg/webhook/utils.go:73]: Similar but different!
https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go#L118

Comment (kumare3) [reply pkg/webhook/utils.go:73]: should we move them all to pluginmachinery?

**Review by kumare3 — APPROVED** [review]

**Pull request merged**